File: shared/src/main/java/ru/privatenull/pnauth/api/AuthAdminApi.java
package ru.privatenull.pnauth.api;

import java.util.concurrent.CompletableFuture;

/** Administrative account operations. */
public interface AuthAdminApi {
    CompletableFuture<AuthResult> unregister(String username);
    CompletableFuture<AuthResult> adminChangePassword(String username, String newPassword);
    CompletableFuture<AuthResult> forceRegister(String username, String password);
    CompletableFuture<AuthResult> forceLogin(String username);
    CompletableFuture<AuthResult> togglePremium(String username);
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/api/AuthAdminApi.kt
package ru.privatenull.pnauth.api

import java.util.concurrent.CompletableFuture

/** Administrative account operations. */
interface AuthAdminApi {
    fun unregister(username: String): CompletableFuture<AuthResult>
    fun adminChangePassword(username: String, newPassword: String): CompletableFuture<AuthResult>
    fun forceRegister(username: String, password: String): CompletableFuture<AuthResult>
    fun forceLogin(username: String): CompletableFuture<AuthResult>
    fun togglePremium(username: String): CompletableFuture<AuthResult>
}
